docroot/modules/custom/mass_translations/mass_translations.module

```diff
@@ -112,8 +112,13 @@ function mass_translations_element_info_alter(array &$info) {
  * Implements hook_pathauto_alias_alter().
  */
 function mass_translations_pathauto_alias_alter(&$alias, array &$context) {
-  // Set language of all aliases to undefined to allow aliases to remain independent of language.
-  $context['language'] = 'en';
+  // For nodes, set the default language to undefined to allow aliases to be used in canonical links.
+  if ($context['module'] == 'node' && $context['language'] != 'en') {
+    $context['language'] = LanguageInterface::LANGCODE_NOT_SPECIFIED;
+  } else {
+    // Set default pathauto alias language to English to prevent duplicate aliases.
```

Comment:
```suggestion
    // Set the context pathauto alias language to English to prevent duplicate aliases. This is especially important for documents.
```